infinitic-storage-mysql/src/main/kotlin/io/infinitic/storage/mysql/MySQLKeyCounterStorage.kt

```diff
@@ -0,0 +1,94 @@
+/**
+ * "Commons Clause" License Condition v1.0
+ *
+ * The Software is provided to you by the Licensor under the License, as defined
+ * below, subject to the following condition.
+ *
+ * Without limiting other conditions in the License, the grant of rights under the
+ * License will not include, and the License does not grant to you, the right to
+ * Sell the Software.
+ *
+ * For purposes of the foregoing, “Sell” means practicing any or all of the rights
+ * granted to you under the License to provide to third parties, for a fee or
+ * other consideration (including without limitation fees for hosting or
+ * consulting/ support services related to the Software), a product or service
+ * whose value derives, entirely or substantially, from the functionality of the
+ * Software. Any license notice or attribution required by the License must also
+ * include this Commons Clause License Condition notice.
+ *
+ * Software: Infinitic
+ *
+ * License: MIT License (https://opensource.org/licenses/MIT)
+ *
+ * Licensor: infinitic.io
+ */
+
+package io.infinitic.storage.mysql
+
+import com.zaxxer.hikari.HikariDataSource
+import io.infinitic.common.storage.keyCounter.KeyCounterStorage
+import org.jetbrains.annotations.TestOnly
+
+private const val MYSQL_TABLE = "key_counter_storage"
+class MySQLKeyCounterStorage(
+    private val pool: HikariDataSource
+) : KeyCounterStorage {
+
+    companion object {
+        fun of(config: MySQL) = MySQLKeyCounterStorage(getPool(config))
+    }
+
+    init {
+        // Create MySQL table at init, for first time usage
+        pool.connection.use {
+            it.prepareStatement(
+                "CREATE TABLE IF NOT EXISTS $MYSQL_TABLE (" +
+                    "`id` INT AUTO_INCREMENT PRIMARY KEY," +
+                    "`key` VARCHAR(255) NOT NULL UNIQUE," +
```

Comment:
Question why not use `key` directly as a primary key of the table?

Reply:
Great question I did this at first, but after a little bit of research it seems like the "MySQL way" is more about having auto_incremented int as Primary keys. 
https://stackoverflow.com/questions/19299874/can-i-use-varchar-as-the-primary-key

Also in my understanding InnoDB links other indexes with the primary key, but if it's not an integer it will create an "under the hood" ID that's an int to link the primary key too (because it performs better).

So I added a primary key that's an auto incremented INT for the MySQL & InnoDB convention kind of. 

Of course it's disputable & we can make they string key our primary key